### PR TITLE
[IMP] sales: 100 percent down

### PR DESCRIPTION
**Pull request by pabr-odoo**
Task [3971833](https://www.odoo.com/odoo/project/3835/tasks/3971833?cids=3)
Added a new section to the Down payments doc explaining the process of using a 100% down payment option when invoicing.

Related Task [3928233](https://www.odoo.com/odoo/project/49/tasks/3928233?cids=3)

### CODE REVIEW

**Comment by pabr-odoo** [comment]
Hey @StraubCreative, tagging you on this new section on using 100% down that you mentioned before. Thanks

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:145]:
```suggestion
Doing so reveals a :guilabel:`Register Payment` pop-up form, which is auto-populated with the
```
While it totally seems like a single word, a hyphen is actually needed in this instance... https://www.dictionary.com/browse/auto-populate

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:192]:
```suggestion
Once again, the :guilabel:`Register Payment` appears, with all fields auto-populated with the
```
same reasoning as above ☝️

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:217]:
```suggestion
   This is because, due to the invoicing policy, the product(s) would have to be delivered *before*
```
SUGGESTION: since there could be multiple products that need to be delivered, I believe using this parenetical is apt. However, this is totally your call 👍

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:234]:
```suggestion
down payment <sales/invoicing/down_payment/50-percent-down-payments>`, but with fewer steps. 

The following example focuses on a *Solar Panel Installation* product.
```
OPTIONAL: added comma and broke up sentences for enhanced sentence clarity and flow

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:236]:
```suggestion
To configure a 100% down payment, begin by navigating to :menuselection:`Sales app --> New`, and add
```
SUGGESTION: adding a comma for enhanced sentence flow/pacing

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:243]:
```suggestion
Upon clicking the :guilabel:`Confirm` button, the quotation turns into a sales order. At that point, an invoice can now be created by clicking :guilabel:`Create Invoice` in the top-left corner.
```
SUGGESTION: some added clarity

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:250]:
```suggestion
(percentage)`, and type `100` in the :guilabel:`Down Payment Amount` field. Then, if desired, select an :guilabel:`Income Account' and :guilabel:`Customer Taxes` in their respective fields.
```
Just some added information to round out the fields mentioned in the 'Create invoice' pop-up window

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:256]:
```suggestion
Next, click :guilabel:`Create Draft Invoice` to create an invoice draft. This will also
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:257]:
```suggestion
bring the draft invoice into view, which includes the :guilabel:`Down payment` as a :guilabel:`Product` in the
```
added a GUI to 'Down payment' because it appears on screen

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:262]: During my revisions, using a standard 17.0 runbot, I didn't see a series of smart buttons appear at the top of the page. So, that is worth a quick double-check to ensure the accuracy of that statement. 

UPDATE: I can see on line 268 that you refer to the 'Register Payment' button as a "smart button." That is actually not the case. Totally easy mistake to make, though. The "smart buttons" are the buttons that appear at the top of the page (typically white boxes with figures, icons, etc.) -- the purple and grey buttons that appear on the menu line, slightly above the form are just regular buttons.

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:274]:
```suggestion
Doing so reveals a :guilabel:`Register Payment` pop-up form, which is auto-populated with the
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:275]:
```suggestion
necessary information. Confirm the information provided is correct, and make any necessary
```
OPTIONAL: added comma for enhanced sentence clarity/flow

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (ksc-odoo) [inline content/applications/sales/sales/invoicing/down_payment.rst:289]:
```suggestion
The process is now complete, and the 100% down payment has been successfully applied.
```

**Review by ksc-odoo — APPROVED** [review]
Hey @pabr-odoo -- just finished my Review of this PR. Great work, man. 👍 Just have a handful of comments/feedbacks/suggestions that require your attention -- but I am approving now because I don't anticipate you having any issues. However, if you have any questions about any of them (or how to implement them), don't hesitate to reach out.
Once you make the required adjustments/modifications, feel free to tag this for Tech Review. Thanks! 👍

**Comment by pabr-odoo** [comment]
Hey @samueljlieber this additional section is ready for tech review once you get the chance. Thanks!

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:85]: Small update to only include the parent folder in the custom anchor convention: 
`app or module name/(optional parent folder)/section heading`
```suggestion
.. _sales/invoicing/50-percent-down-payments:
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:230]:
```suggestion
.. _sales/invoicing/100-percent-down-payments:
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:236]: Update ref tag after updating custom anchor
```suggestion
down payment <sales/invoicing/50-percent-down-payments>`, but with fewer steps.
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:242]: Since you are referencing the exact UI button, please match it's case :)
```suggestion
   be generated, and **will not** display the *Create Invoice* button on the Sales Order.
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:244]:
```suggestion
   Following this example **will** cause the *Create Invoice* button to be displayed on the Sales
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:247]: Nitpick- remove extra line :)
```suggestion
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:255]: Nitpick- since the user has to type in the product name, lets use literal formatting here :)
```suggestion
`Solar Panel Installation` product.
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:263]: To try and cut down on the number of images in this doc, I think we can remove most of these. Especially since the process is so similar to the 50% DP example, and your written instructions are clear. If you feel like these images are important to keep, please feel free to do so 👍
```suggestion
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:266]: guilabel tag fix
```suggestion
an :guilabel:`Income Account` and :guilabel:`Customer Taxes` in their respective fields.
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:267]: It is helpful to include this note (from the 50% DP section) that addresses these fields, as I was confused why they did not appear for me at this step :)
```suggestion

.. note::
   The :guilabel:`Income Account` and :guilabel:`Customer Taxes` fields are *not* required fields,
   and they will *not* appear if they've already been preconfigured in previous down payment
   requests.

   For more information, check out the documentation on :ref:`customer taxes modification on down
   payments <sales/invoicing/customer-tax-modification-down-payments>` and :ref:`income account
   modification on down payments <sales/invoicing/income-account-modification>`.
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:273]: typo
```suggestion
bring the draft invoice into view, which includes the :guilabel:`Down payment` as a
```

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:296]: Same reasoning as above, your written instructions are very clear– these images can be removed IMO :)
```suggestion

The payment can be registered by clicking the :guilabel:`Register Payment` button.

Doing so reveals a :guilabel:`Register Payment` pop-up form, which is auto-populated with the
necessary information. Confirm the information provided is correct and make any necessary
adjustments. When ready, click the :guilabel:`Create Payment` button.

```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @pabr-odoo! Nice work on this Sales PR on 100% down payments. Your changes are looking good to me! I just have a couple quick fixes, nitpicks, and suggestions for you. Please tag me once more for a final look once these have been addressed, thank you for your work!

**Comment by pabr-odoo** [comment]
Hey @samueljlieber thanks so much for your review. I looked over and committed your suggestions. Please give it a final look when you get the chance and let me know if you have any other changes. Thanks!

File: content/applications/sales/sales/invoicing/down_payment.rst
Comment (samueljlieber) [inline content/applications/sales/sales/invoicing/down_payment.rst:271]: Need an empty line between these two directives :)
```suggestion
   modification on down payments <sales/invoicing/income-account-modification>`.
   
.. image:: down_payment/100p-down-payment-percentage.png
```

**Review by samueljlieber — APPROVED** [review]
Hi @pabr-odoo, thank you for implementing my suggestions! Changes look good 👍 Approving with one more quick fix. Please make this fix before merging :)
..
@robodoo delegate=pabr-odoo

**Comment by pabr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
